compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -286,11 +286,18 @@ static Value buildReductionStrategyThreadDistributionPart(
   return variantH;
 }
 
+/// Structure to hold the parameters related to GPU reduction strategy.
+struct GPUReductionStrategyInfos {
```

Comment:
Fly by comment, feel free to ignore. I am personally not a fan of having such structs.... Would be better to have an actually `Attribute` that carries the same information... otherwise this gets harder to debug...

Reply:
You mean an attribute in the IR? The idea is that the transform dialect will be the IR serialized version of those information. The structure is just an easier way to communicate between the matcher and the transform dialect builder. I don't know if we want to serialize information between the matcher and the transform dialect builder. My gut feeling is that it is not worth it.
Happy to discuss it more. We will need to get to an agreement by the end of the bring-up

Reply:
I see your point. Happy to wait and see how this evolves.

Reply:
Longer-term, these will likely be materialized as attributes of a transform dialect "expert" op, or some other sort of parameterization of the transform IR. Shorter-term, I see this as a readability improvement over the flat `TileSizesListType` where we have magic index 0 corresponding to workgroup tile sizes, magic index 1 corresponding to fill op tile sizes, etc.

Reply:
+1

Reply:
> Longer-term, these will likely be materialized as attributes of a transform dialect "expert" op, or some other sort of parameterization of the transform IR.

Sounds good. Thats what I was hoping for. 

> magic index 1 corresponding to fill op tile sizes, etc.

Not really sure what that is, but probably doesnt matter. There is actually no "universal magic index". Depending on what the pipeline needs, you set that many tile sizes.. It is just a bucket of values that are set during `KernelConfig.cpp` and used within the pipeline. The only "magic index" happens to be 0 for workgroup tile sizes since all backends first using tiling to distribute the tiles to workgroups. This all predates `AttributeInterfaces` (or at least my understanding of `AttributeInterfaces`, so this was a way to not have an explosion in number of attributes.